esmvalcore/_recipe_checks.py

```diff
@@ -90,10 +91,25 @@ def variable(var, required_keys):
                 missing, var.get('short_name'), var.get('diagnostic')))
 
 
-def data_availability(input_files, var):
+def data_availability(input_files, var, dirnames, filenames):
     """Check if the required input data is available."""
     if not input_files:
-        raise RecipeError("No input files found for variable {}".format(var))
+        var.pop('filename', None)
+        logger.error("No input files found for variable %s", var)
+        if dirnames and filenames:
+            patterns = itertools.product(dirnames, filenames)
+            patterns = [os.path.join(d, f) for (d, f) in patterns]
+            logger.error("Looked for files matching\n%s", "\n".join(patterns))
```

Comment:
```suggestion
            logger.error("Looked for files matching: %s", "\n".join(patterns))
```

Reply:
I agree that this is nicer if `patterns` only contains one element, but if it contains more it's not really pretty. I change it to respect both cases.